src/back/glsl/mod.rs

```diff
@@ -1187,30 +1192,18 @@ impl<'a, W: Write> Writer<'a, W> {
             // This is where we can generate intermediate constants for some expression types.
             Statement::Emit(ref range) => {
                 for handle in range.clone() {
+                    if let Some(name) = ctx.named_expressions.get(&handle) {
+                        write!(self.out, "{}", INDENT.repeat(indent))?;
+                        let sanitized_name = self.namer.sanitize(name);
```

Comment:
this turned out to be pretty nice! Now there isn't a concern that the local map partially duplicates `named_expressions`, since in fact they may be different